"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import typing
import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor = ...

class FeedErrorEnum(google.protobuf.message.Message):
    """Proto file describing feed errors.

    Container for enum describing possible feed errors.
    """
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    class _FeedError:
        ValueType = typing.NewType('ValueType', builtins.int)
        V: typing_extensions.TypeAlias = ValueType
    class _FeedErrorEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[_FeedError.ValueType], builtins.type):
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
        UNSPECIFIED: FeedErrorEnum.FeedError.ValueType = ...  # 0
        """Enum unspecified."""

        UNKNOWN: FeedErrorEnum.FeedError.ValueType = ...  # 1
        """The received error code is not known in this version."""

        ATTRIBUTE_NAMES_NOT_UNIQUE: FeedErrorEnum.FeedError.ValueType = ...  # 2
        """The names of the FeedAttributes must be unique."""

        ATTRIBUTES_DO_NOT_MATCH_EXISTING_ATTRIBUTES: FeedErrorEnum.FeedError.ValueType = ...  # 3
        """The attribute list must be an exact copy of the existing list if the
        attribute ID's are present.
        """

        CANNOT_SPECIFY_USER_ORIGIN_FOR_SYSTEM_FEED: FeedErrorEnum.FeedError.ValueType = ...  # 4
        """Cannot specify USER origin for a system generated feed."""

        CANNOT_SPECIFY_GOOGLE_ORIGIN_FOR_NON_SYSTEM_FEED: FeedErrorEnum.FeedError.ValueType = ...  # 5
        """Cannot specify GOOGLE origin for a non-system generated feed."""

        CANNOT_SPECIFY_FEED_ATTRIBUTES_FOR_SYSTEM_FEED: FeedErrorEnum.FeedError.ValueType = ...  # 6
        """Cannot specify feed attributes for system feed."""

        CANNOT_UPDATE_FEED_ATTRIBUTES_WITH_ORIGIN_GOOGLE: FeedErrorEnum.FeedError.ValueType = ...  # 7
        """Cannot update FeedAttributes on feed with origin GOOGLE."""

        FEED_REMOVED: FeedErrorEnum.FeedError.ValueType = ...  # 8
        """The given ID refers to a removed Feed. Removed Feeds are immutable."""

        INVALID_ORIGIN_VALUE: FeedErrorEnum.FeedError.ValueType = ...  # 9
        """The origin of the feed is not valid for the client."""

        FEED_ORIGIN_IS_NOT_USER: FeedErrorEnum.FeedError.ValueType = ...  # 10
        """A user can only create and modify feeds with USER origin."""

        INVALID_AUTH_TOKEN_FOR_EMAIL: FeedErrorEnum.FeedError.ValueType = ...  # 11
        """Invalid auth token for the given email."""

        INVALID_EMAIL: FeedErrorEnum.FeedError.ValueType = ...  # 12
        """Invalid email specified."""

        DUPLICATE_FEED_NAME: FeedErrorEnum.FeedError.ValueType = ...  # 13
        """Feed name matches that of another active Feed."""

        INVALID_FEED_NAME: FeedErrorEnum.FeedError.ValueType = ...  # 14
        """Name of feed is not allowed."""

        MISSING_OAUTH_INFO: FeedErrorEnum.FeedError.ValueType = ...  # 15
        """Missing OAuthInfo."""

        NEW_ATTRIBUTE_CANNOT_BE_PART_OF_UNIQUE_KEY: FeedErrorEnum.FeedError.ValueType = ...  # 16
        """New FeedAttributes must not affect the unique key."""

        TOO_MANY_ATTRIBUTES: FeedErrorEnum.FeedError.ValueType = ...  # 17
        """Too many FeedAttributes for a Feed."""

        INVALID_BUSINESS_ACCOUNT: FeedErrorEnum.FeedError.ValueType = ...  # 18
        """The business account is not valid."""

        BUSINESS_ACCOUNT_CANNOT_ACCESS_LOCATION_ACCOUNT: FeedErrorEnum.FeedError.ValueType = ...  # 19
        """Business account cannot access Business Profile."""

        INVALID_AFFILIATE_CHAIN_ID: FeedErrorEnum.FeedError.ValueType = ...  # 20
        """Invalid chain ID provided for affiliate location feed."""

        DUPLICATE_SYSTEM_FEED: FeedErrorEnum.FeedError.ValueType = ...  # 21
        """There is already a feed with the given system feed generation data."""

        GMB_ACCESS_ERROR: FeedErrorEnum.FeedError.ValueType = ...  # 22
        """An error occurred accessing Business Profile."""

        CANNOT_HAVE_LOCATION_AND_AFFILIATE_LOCATION_FEEDS: FeedErrorEnum.FeedError.ValueType = ...  # 23
        """A customer cannot have both LOCATION and AFFILIATE_LOCATION feeds."""

        LEGACY_EXTENSION_TYPE_READ_ONLY: FeedErrorEnum.FeedError.ValueType = ...  # 24
        """Feed-based extension is read-only for this extension type."""

    class FeedError(_FeedError, metaclass=_FeedErrorEnumTypeWrapper):
        """Enum describing possible feed errors."""
        pass

    UNSPECIFIED: FeedErrorEnum.FeedError.ValueType = ...  # 0
    """Enum unspecified."""

    UNKNOWN: FeedErrorEnum.FeedError.ValueType = ...  # 1
    """The received error code is not known in this version."""

    ATTRIBUTE_NAMES_NOT_UNIQUE: FeedErrorEnum.FeedError.ValueType = ...  # 2
    """The names of the FeedAttributes must be unique."""

    ATTRIBUTES_DO_NOT_MATCH_EXISTING_ATTRIBUTES: FeedErrorEnum.FeedError.ValueType = ...  # 3
    """The attribute list must be an exact copy of the existing list if the
    attribute ID's are present.
    """

    CANNOT_SPECIFY_USER_ORIGIN_FOR_SYSTEM_FEED: FeedErrorEnum.FeedError.ValueType = ...  # 4
    """Cannot specify USER origin for a system generated feed."""

    CANNOT_SPECIFY_GOOGLE_ORIGIN_FOR_NON_SYSTEM_FEED: FeedErrorEnum.FeedError.ValueType = ...  # 5
    """Cannot specify GOOGLE origin for a non-system generated feed."""

    CANNOT_SPECIFY_FEED_ATTRIBUTES_FOR_SYSTEM_FEED: FeedErrorEnum.FeedError.ValueType = ...  # 6
    """Cannot specify feed attributes for system feed."""

    CANNOT_UPDATE_FEED_ATTRIBUTES_WITH_ORIGIN_GOOGLE: FeedErrorEnum.FeedError.ValueType = ...  # 7
    """Cannot update FeedAttributes on feed with origin GOOGLE."""

    FEED_REMOVED: FeedErrorEnum.FeedError.ValueType = ...  # 8
    """The given ID refers to a removed Feed. Removed Feeds are immutable."""

    INVALID_ORIGIN_VALUE: FeedErrorEnum.FeedError.ValueType = ...  # 9
    """The origin of the feed is not valid for the client."""

    FEED_ORIGIN_IS_NOT_USER: FeedErrorEnum.FeedError.ValueType = ...  # 10
    """A user can only create and modify feeds with USER origin."""

    INVALID_AUTH_TOKEN_FOR_EMAIL: FeedErrorEnum.FeedError.ValueType = ...  # 11
    """Invalid auth token for the given email."""

    INVALID_EMAIL: FeedErrorEnum.FeedError.ValueType = ...  # 12
    """Invalid email specified."""

    DUPLICATE_FEED_NAME: FeedErrorEnum.FeedError.ValueType = ...  # 13
    """Feed name matches that of another active Feed."""

    INVALID_FEED_NAME: FeedErrorEnum.FeedError.ValueType = ...  # 14
    """Name of feed is not allowed."""

    MISSING_OAUTH_INFO: FeedErrorEnum.FeedError.ValueType = ...  # 15
    """Missing OAuthInfo."""

    NEW_ATTRIBUTE_CANNOT_BE_PART_OF_UNIQUE_KEY: FeedErrorEnum.FeedError.ValueType = ...  # 16
    """New FeedAttributes must not affect the unique key."""

    TOO_MANY_ATTRIBUTES: FeedErrorEnum.FeedError.ValueType = ...  # 17
    """Too many FeedAttributes for a Feed."""

    INVALID_BUSINESS_ACCOUNT: FeedErrorEnum.FeedError.ValueType = ...  # 18
    """The business account is not valid."""

    BUSINESS_ACCOUNT_CANNOT_ACCESS_LOCATION_ACCOUNT: FeedErrorEnum.FeedError.ValueType = ...  # 19
    """Business account cannot access Business Profile."""

    INVALID_AFFILIATE_CHAIN_ID: FeedErrorEnum.FeedError.ValueType = ...  # 20
    """Invalid chain ID provided for affiliate location feed."""

    DUPLICATE_SYSTEM_FEED: FeedErrorEnum.FeedError.ValueType = ...  # 21
    """There is already a feed with the given system feed generation data."""

    GMB_ACCESS_ERROR: FeedErrorEnum.FeedError.ValueType = ...  # 22
    """An error occurred accessing Business Profile."""

    CANNOT_HAVE_LOCATION_AND_AFFILIATE_LOCATION_FEEDS: FeedErrorEnum.FeedError.ValueType = ...  # 23
    """A customer cannot have both LOCATION and AFFILIATE_LOCATION feeds."""

    LEGACY_EXTENSION_TYPE_READ_ONLY: FeedErrorEnum.FeedError.ValueType = ...  # 24
    """Feed-based extension is read-only for this extension type."""


    def __init__(self,
        ) -> None: ...
global___FeedErrorEnum = FeedErrorEnum
